core/src/main/java/com/linecorp/armeria/common/util/AbstractListenable.java

```diff
@@ -40,7 +41,7 @@
     private final Set<Consumer<? super T>> updateListeners =
             new ObjectLinkedOpenCustomHashSet<>(IdentityHashStrategy.of());
 
-    private final ReentrantLock reentrantLock = new ReentrantLock();
+    private final ReentrantLock reentrantLock = new ReentrantShortLock();
```

Comment:
It should be an abnormal case if we use `ReentrantLock` for a long lock.
What do you think of adding a check style rule to early detect `ReentrantLock` allocations?
For example,
```xml
<module name="RegexpSingleline">
  <property name="id" value="ReentrantLock"/>
  <property name="format" value="new ReentrantLock()"/>
  <property name="message" value="Consider using ReentrantShortLock."/>
</module>
```